src/platforms/common/stm32/rtt_if.c

```diff
@@ -58,6 +58,17 @@ inline static bool recv_set_nak()
 	return recv_bytes_free() < 2U * CDCACM_PACKET_SIZE;
 }
 
+void rtt_load_recv_buf(const char *data_buf, uint16_t len)
+{
+	/* copy data to recv_buf */
+	for (int i = 0; i < len; i++) {
```

Comment:
This isn't actually your fault, but the use of raw `int` introduces a signed-unsigned conversion and a silent error here. Please switch this to use `size_t` or at least `uint16_t` to match the `len` argument.

Similarly with the proper capitalisation of comments. We know that one's also not your fault here, this code has just not yet been cleaned up and the original author was distinctly lazy about it.